File: aoc/2024/01/pt1.py
from heapq import heappush, heappop
from aoc import read


def main():
    total = 0

    a_list = []
    b_list = []
    for a, b in read(lambda x: map(int, x.split())):
        heappush(a_list, a)
        heappush(b_list, b)
    
    while a_list:
        a, b = heappop(a_list), heappop(b_list)
        total += abs(a - b)

    return total


if __name__ == '__main__':
    print(main())

File: aoc/y2024/d01/pt1.ts
import { sorted_insert } from '../../index';


const [aList, bList] = [[], []];

for await (const line of console) {
  const [a, b] = line.split(/\s+/).map(x => parseInt(x, 10));

  sorted_insert(aList, a);
  sorted_insert(bList, b);
}

const total = aList.reduce(
  (acc, n, i) => acc + Math.abs(n - bList[i]), 0
);

console.log(total);